lib/orbit/evented.js

```diff
@@ -191,72 +191,36 @@ var Evented = {
       return listeners;
     },
 
-    resolve: function(eventNames) {
-      var listeners = this.listeners(eventNames),
-          args = Array.prototype.slice.call(arguments, 1);
-
-      return new Orbit.Promise(function(resolve, reject) {
-        var resolveEach = function() {
-          if (listeners.length === 0) {
-            reject();
-          } else {
-            var listener = listeners.shift();
-            var response = listener[0].apply(listener[1], args);
-
-            if (response) {
-              response.then(
-                function(success) {
-                  resolve(success);
-                },
-                function(error) {
-                  resolveEach();
-                }
-              );
-            } else {
-              resolveEach();
-            }
-          }
-        };
-
-        resolveEach();
-      });
+    resolve(eventNames, ...args) {
```

Comment:
Bluebird calls `resolve` `any` http://bluebirdjs.com/docs/api/promise.any.html which seems like a good option. I did wonder if we should just replace `resolve` with Promise.all but the original implementation specifically chained the promises in sequence and I'm pretty sure Promise.all allows parallel execution. Not sure if this is important or not @dgeb?